src/engraving/libmscore/select.cpp

```diff
@@ -527,7 +527,7 @@ void Selection::appendChord(Chord* chord)
             if (sp->endElement()->isNote()) {
                 Note* endNote = toNote(sp->endElement());
                 Segment* s = endNote->chord()->segment();
-                if (s->tick() < tickEnd()) {
+                if (s && s->tick() < tickEnd()) {
```

Comment:
This particular crash happens here

Reply:
Alternative fix: `if (!s || s->tick() < tickEnd()) {`, avopids the crash too. Which would be more correct?

Reply:
I'd be more worried why a chord would ever not have a segment?

Reply:
Well, then check the sample score from that forum post.

Reply:
Yes, this is curing the symptom rather than the cause. But that isn't a bad thing, or is it?

Reply:
It is only happening directly after the XML import, not anymore once the score has been saved as mscz (and a close/reopen).

Reply:
Well the issue is as you noted it's not clear what the right behaviour is in this case if a chord doesn't have a segment, and it seems like the sort of thing that might have consequences elsewhere.  I'd be trying to fix it in the XML importer first, that's all.
I'd also note that MS codebase, and even the short sample above, would probably be twice as big if it had to check every pointer first, due to the lack of a null propagation operator in C++.

Reply:
Don't check for error conditions you can't handle... but do check those you can.

I've now tried both methods on the xml from the forum post: the resulting scores are 100% identical.

Reply:
But yes, of course it'd be better to fix the XML import, I just don't know where to look. @lvinken ?

Reply:
Indeed, MusicXML import should not result in invalid data. Will try to find the root cause.

Reply:
See comment added to https://musescore.org/en/node/330595 for possible root cause, shouldn't be too hard to track down and fix in a structural way.